otx/algorithms/segmentation/tasks/inference.py

```diff
@@ -187,7 +187,12 @@ def _init_recipe(self):
         logger.info(f"train type = {self._train_type}")
 
         if self._train_type in RECIPE_TRAIN_TYPE:
-            recipe = os.path.join(recipe_root, RECIPE_TRAIN_TYPE[self._train_type])
+            if self._train_type == TrainType.INCREMENTAL and self._hyperparams.learning_parameters.enable_supcon:
+                recipe = os.path.join(recipe_root, "supcon.py")
+                if "supcon" not in self._model_dir:
```

Comment:
Is this check necessary? I think we always need to change our model directory to template_dir/supcon_model_dir

Reply:
I often occur the error that `self._model_dir` for training was remained until evaluation and `supcon` was added one more and `self._model_dir` became `.../supcon/supcon` whenever I ran the test.
SupCon for cls uses the similar condition to avoid overlapped path update.
https://github.com/openvinotoolkit/training_extensions/blob/8894d3d811ce1c770af3c1e3e74ec825a9ba16af/otx/algorithms/classification/tasks/inference.py#L380-L388
If there is another way to handle it, I can update it. Could you recommend it?

Reply:
I glanced supcon support including classification, I feel like supcon is supported as train type already even if it has enable_supcon arg itself. (which has its own recipe file but the contents are almost same as incremental) I understand that model and data pipeline is totally different so it must be reloaded here but about other things, I personally recommend to refine all config for supcon in stage such as making _configure_supcon methods, injecting enable_supcon cfg in the incremental recipe file without making a recipe file. I think we need some discussion about that. But as of now, it's okay to handle these things (and consider classification supcon as well) after merging this PR.